src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1520,51 +1542,41 @@ private String quoteUnixCredentials(String str) {
         return str.replace("'", "'\\''");
     }
 
-    private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".bat");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"");
-            w.flush();
-        }
-        ssh.setExecutable(true);
+    private FilePath createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTextTempFile("pass", ".bat",
+            "echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"", false);
         return ssh;
     }
 
-    private File createUnixSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".sh");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("#!/bin/sh");
-            w.println("echo '" + quoteUnixCredentials(Secret.toString(sshUser.getPassphrase())) + "'");
-        }
-        ssh.setExecutable(true);
+    private FilePath createUnixSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTextTempFile("pass", ".sh",
```

Comment:
this smells insecure.  use [`WorkspaceList.tmpDir(workspace)`](https://github.com/jenkinsci/jenkins/commit/307bfc17385c14aefcb623ccc3e7144e4f01e2a3)

If you can not bump a version to that write a helper that does the same thing.